### PR TITLE
ROC-6377-Updating response time and more time requested reviewing paper response

### PR DESCRIPTION
**Pull request by karthikkannankainos**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-3724
&& 
https://tools.hmcts.net/jira/browse/ROC-6377

### Change description ###

Updating response time and more time requested reviewing paper response

### CODE REVIEW

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4865314&bid=16135444)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4865314&bid=16135444)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  4
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4865314&bid=16135444)

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    @Override
+    public List<Role> getSupportedRoles() {
+        return ROLES;
+    }
+
+    private AboutToStartOrSubmitCallbackResponse updateResponseDeadline(CallbackParams callbackParams) {
+        CallbackRequest callbackRequest = callbackParams.getRequest();
+        Claim claim = toClaimAfterEvent(callbackRequest);
+
+        if (verifyNoDuplicateMoreTimeRequested(callbackRequest)) {
```
Comment (KiranVarma02) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java]: as i was saying in DM just as suggestion, could we maybe extract `verifyNoDuplicateMoreTimeRequested` and all related code to other helper class!!

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4865314&bid=16135444)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4865314&bid=16135444)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4865314&bid=16135444)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4865314&bid=16135444)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  5
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4865314&bid=16135444)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  5
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4865314&bid=16135444)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandlerTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4865314&bid=16135444)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandler.java  5
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/response/PaperResponseReviewedCallbackHandlerTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4865314&bid=16135444)

**Pull request merged**